src/clients/meta/MetaClient.cpp

```diff
@@ -2414,6 +2414,12 @@ folly::Future<StatusOr<bool>> MetaClient::heartbeat() {
     } else {
       req.set_leader_partIds(std::move(leaderIds));
     }
+
+    kvstore::DiskPaths diskPaths;
```

Comment:
You could check if the relation between disk and partition has been updated or not here. See around line 2406.